src/__tests__/paxfulapi_refresh_test.ts

```diff
@@ -0,0 +1,81 @@
+import fetch from 'node-fetch';
+
+// We need this import to get the extra jest assertions
+
+import 'fetch-mock-jest';
+import usePaxful from "../index";
+import { mock } from "jest-mock-extended";
+import { CredentialStorage } from "../oauth";
+import { v4 as UUID } from "uuid";
+
+const fetchMock = require('node-fetch')
+
+jest.mock(
+    'node-fetch',
+    () => require('fetch-mock-jest').sandbox(),
+);
+
+
+const credentials = {
+    clientId: UUID(),
+    clientSecret: UUID(),
+    redirectUri: "callback"
+};
+const credentialStorage = mock<CredentialStorage>();
+
+describe("With the Paxful API SDK", function () {
+    beforeEach(() => fetchMock.reset());
+
+    it("Client credentials reset flow", async () => {
+        fetchMock.once({
+            url: /oauth2\/userinfo/,
+            method: "GET"
+        }, {
+            status: 401,
+            body: ""
+        });
+
+        fetchMock.once({
+            name: 'correct_access_token',
+            url: /oauth2\/token/,
+            method: "POST"
+        }, {
+            status: 200,
+            body: JSON.stringify({
+                access_token: "abc",
+                refresh_token: null,
+                expires_in: 100
+            })
+        }, {
+            sendAsJson: false
+        });
+
+        fetchMock.once({
+            name: 'correct_access_token',
+            url: /oauth2\/userinfo/,
+            method: "GET"
+        }, {
+            status: 200,
+            body: JSON.stringify({ some: "lala" })
+        });
+
+        credentialStorage.getCredentials.mockReturnValue({
+            ...{
+                accessToken: UUID(),
+                refreshToken: UUID(),
+            },
+            expiresAt: new Date()
+        });
+
+        const paxfulApi = usePaxful(credentials, credentialStorage);
+        const profile = await paxfulApi.getProfile();
```

Comment:
getProfile() is only one use use, we need to make sure that at very least invoke() also relies on the same logic (but we also have get(), post() etc methods that you should the same).

Either we have integration tests for each of those scenarios OR create a unit test for RefreshIfNeeded command and then just validate its invocation in all of the said scenarios.

Reply:
Let it be integration tests at the moment. Big chance that we will refactor that part in the future.
Added tests for all other functions invoking HTTP request